src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/abstract_port_implementation.h

```diff
@@ -17,7 +17,7 @@ class AbstractPortImplementation {
   int Start();
   int Join();
   std::vector<int> GetShape();
-  std::vector<PortPtr> GetPorts();
+  virtual std::vector<PortPtr> GetPorts();
 
   DataType dtype_;
```

Comment:
This data type and data shape could be tranformed following data structure of numpy array.